pkg/external_repos/introspect_test.go

```diff
@@ -230,10 +241,59 @@ func TestUpdateIntrospectionStatusMetadata(t *testing.T) {
 				LastIntrospectionSuccessTime: &timestamp,
 				LastIntrospectionError:       pointy.String(""),
 				Status:                       config.StatusValid,
+				PackageCount:                 *pointy.Int(100),
+			},
+		},
+		{
+			name: "Status remains as Invalid",
+			given: TestCaseGiven{
+				status: config.StatusInvalid,
+				count:  1,
+				err:    fmt.Errorf("Error remains, keep it as Invalid"),
+			},
+			expected: dao.Repository{
+				LastIntrospectionTime:        &timestamp,
+				LastIntrospectionUpdateTime:  nil,
+				LastIntrospectionSuccessTime: nil,
```

Comment:
Since these are nil by default, I think it's okay to leave them out.

Reply:
there are a couple of other places where the nils are being explicitly defined, so I left two more comments to point them out.